### PR TITLE
Recognize orientation changes and provide new dimensions

### PR DESCRIPTION
**Pull request by donni106**
Used https://docs.expo.io/versions/latest/sdk/screen-orientation to implement a listener for orientation changes. Please see commits for further information.

Updated the `ImageCarousel` component to respond to orientation changes.

iPhone Landscape | iPad Landscape
------------ | -------------
Simulator Screen Shot - iPhone 8 Plus - 2020-09-24 at 15 14 24 | Simulator Screen Shot - iPad Pro (9 7-inch) - 2020-09-24 at 15 14 32

### CODE REVIEW

**Review by marcometz — APPROVED** [review]
cool! Die Anpassungen, die dafür notwendig waren, sehen gar nicht so komplex aus :-)

File: src/OrientationProvider.js [hunk 1]
```diff
@@ -1,28 +1,50 @@
...
+    ScreenOrientation.addOrientationChangeListener(({ orientationInfo }) => {
+      // we need to wait a short period of time before reading and setting the orientation and
+      // dimensions, because some devices may take longer then other to update the orientation.
+      setTimeout(() => {
+        // https://docs.expo.io/versions/latest/sdk/screen-orientation/#screenorientationaddorientationchangelistenerlistener
+        setOrientation(getOrientation(orientationInfo.orientation));
+        setDimensions({
+          height: Dimensions.get('window').height,
+          width: Dimensions.get('window').width
+        });
```
Comment (SoleCincis) [inline src/OrientationProvider.js:46]: we set the state for Orientation and Dimension, using addOrientationChangeListener method (from expo-screen-orientation) we set new Orientation using the previously declared getOrientation function then we set new Dimensions again using  Dimensions.get from RN .

File: src/OrientationProvider.js [hunk 2]
```diff
@@ -31,7 +53,7 @@ export const OrientationProvider = ({ children }) => {
   }, []);
 
   return (
-    <OrientationContext.Provider value={{ dimensions }}>
+    <OrientationContext.Provider value={{ orientation, dimensions }}>
       {children}
     </OrientationContext.Provider>
```
Comment (SoleCincis) [inline src/OrientationProvider.js:58]: OrientationContext.Provider is passing its logic and orientation and dimensions props to all {children} component across the app , all children inside  <MainAppWithApolloProvider />

File: src/helpers/imageHelper.js [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
-export const imageHeight = (horizontal = false) => {
+export const imageHeight = (width) => {
   // image aspect ratio is 360x180, so for accurate ratio in our view we need to calculate
-  // a factor with our current device with for the image, to set a correct height
-  const factor = imageWidth(horizontal) / 360;
+  // a factor with our current device width for the image, to set a correct height
+  const factor = width / 360;
 
   return 180 * factor;
 };
```
Comment (SoleCincis) [inline src/helpers/imageHelper.js:13]: I can't formulate a sentence to describe what's happening here

File: src/components/ImagesCarousel.js [hunk 4]
```diff
@@ -24,23 +24,30 @@ const TouchableImage = ({ navigation, item }) => {
...
+  const { orientation, dimensions } = useContext(OrientationContext);
 
   const renderItem = ({ item }) => {
     const { routeName, params } = item.picture;
 
+    // special image width for carousel, to be not full width on landscape
+    const width = orientation === 'landscape' ? dimensions.height : dimensions.width;
+    const imageStyle = {
+      height: imageHeight(width),
+      width
```
Comment (SoleCincis) [inline src/components/ImagesCarousel.js:48]: imageHeight from imageHelper , what is it happening here ? from comment image will not be full width on landscape , how is this happening ?

File: src/OrientationProvider.js [hunk 5]
```diff
@@ -1,28 +1,50 @@
...
 
   useEffect(() => {
-    ScreenOrientation.addOrientationChangeListener(() => {
-      // https://docs.expo.io/versions/latest/sdk/screen-orientation/#screenorientationaddorientationchangelistenerlistener
-      setDimensions({
-        height: Dimensions.get('window').height,
-        width: Dimensions.get('window').width
-      });
+    ScreenOrientation.addOrientationChangeListener(({ orientationInfo }) => {
+      // we need to wait a short period of time before reading and setting the orientation and
```
[the remark this answers is not in the document]
Comment (donni106) [reply src/OrientationProvider.js:38]: it's from the package and passed from the listener https://docs.expo.io/versions/latest/sdk/screen-orientation/#screenorientationaddorientationchangelistenerlistener
which is a `OrientationChangeEvent` https://docs.expo.io/versions/latest/sdk/screen-orientation/#screenorientationorientationchangeevent
which gives us to attriutes. one is `orientationInfo`

> orientationLock (OrientationLock): The current OrientationLock of the device.
> orientationInfo (ScreenOrientationInfo): The current ScreenOrientationInfo of the device.

File: src/helpers/imageHelper.js (hunk 3 above)
Comment (donni106) [reply src/helpers/imageHelper.js:13]: a number is returned, which is calculated based on width.

if width is 360, than `factor` is 360 / 360 = 1. then the return is 180 * 1 = 180.
this means, if the width is 360, the height will be 180.
we calculate this to ensure that images always be in 2:1 ratio.

File: src/components/ImagesCarousel.js (hunk 4 above)
Comment (donni106) [reply src/components/ImagesCarousel.js:48]: imageStyle is set with the values passed.

height will be the height based on the width calculated in the helper, which i described in a comment before.
width will be the width depending on the orientation, which ensures, that the image has the width of the device in portrait and the height of the device in landscape. width in portrait equals height in landscape. so at eh end, images are always same sized on portrait and landscape, right?!

**Pull request merged**